search/intermediate_result.cpp

```diff
@@ -282,19 +286,38 @@ void FillDetails(FeatureType & ft, Result::Details & details)
 
   feature::TypesHolder const typesHolder(ft);
 
-  details.m_isHotel = ftypes::IsHotelChecker::Instance()(typesHolder);
-  if (details.m_isHotel && strings::to_uint(ft.GetMetadata(feature::Metadata::FMD_STARS), details.m_stars))
-    details.m_stars = std::min(details.m_stars, osm::MapObject::kMaxStarsCount);
-  else
-    details.m_stars = 0;
+  uint8_t starsCount = 0;
+  bool const isHotel = ftypes::IsHotelChecker::Instance()(typesHolder);
+  if (isHotel && strings::to_uint(ft.GetMetadata(feature::Metadata::FMD_STARS), starsCount))
+    starsCount = std::min(starsCount, osm::MapObject::kMaxStarsCount);
+  std::string stars;
+  for (int i = 0; i < starsCount; ++i)
+    stars.append(osm::MapObject::kStarSymbol);
 
   auto const cuisines = feature::GetLocalizedCuisines(typesHolder);
-  details.m_cuisine = strings::JoinStrings(cuisines, osm::MapObject::kFieldsSeparator);
+  auto const cuisine = strings::JoinStrings(cuisines, osm::MapObject::kFieldsSeparator);
 
   auto const roadShields = ftypes::GetRoadShieldsNames(ft);
-  details.m_roadShields = strings::JoinStrings(roadShields, osm::MapObject::kFieldsSeparator);
-
-  details.m_fee = feature::GetLocalizedFeeType(typesHolder);
+  auto const roadShield = strings::JoinStrings(roadShields, osm::MapObject::kFieldsSeparator);
+
+  auto const fee = feature::GetLocalizedFeeType(typesHolder);
```

Comment:
@biodranik @vng What do you think about `auto`? I'm not particularly fond of it in this scenario, but the decision is yours. 😁

Reply:
`auto` is ok if knowing the type is not important or not necessary to understand the logic, or if the type can be obviously guessed.

An explicit type sometimes is clearer and saves time for the reviewer/code reader.